src/views/List.jsx

```diff
@@ -1,15 +1,61 @@
 import { ListItem } from '../components';
+import { useEffect, useState } from 'react';
 
 export function List({ data }) {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [searchResults, setSearchResults] = useState([]);
+	// user type in search item - query
+	// as letters come in, filters items rendered on page
+	// set query to searchQuery
+	// render onto page search Results -> setting whatever is filtered into searchResults
+
+	function filterResults(query) {
+		return data.filter((item) =>
+			item.name.toLowerCase().includes(query.toLowerCase()),
+		);
+	}
+
+	useEffect(() => {
+		setSearchResults(filterResults(searchQuery));
+
+		// ignoring dependency array warning for now
+		// adding filterResults causes infinite re-render
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [searchQuery]);
+
+	function handleClick() {
+		setSearchResults([]);
```

Comment:
For UX and behavior, I think this should be refactored. We don't want an empty list array on search input clear -- we instead want to target the search query for clearing so there's no query to filter and the list resets back to full.

Currently, when I clear the search results, the input field does clear but so does the list. Instead of being able to see the unfiltered list the page hangs in loading for a few seconds and then I see an empty list component.

Reply:
Good catch. I refactored it and set the search query state to an empty string - `setSearchQuery('');`.

Thanks!